client_for is incompatible with pypy

```
tests/tornado/test_hyperbahn.py:62:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tchannel/tornado/tchannel.py:116: in __init__
    self.hooks.register(ZipkinTraceHook(tchannel=self))
tchannel/zipkin/zipkin_trace.py:44: in __init__
    self.tracer = TChannelZipkinTracer(tchannel)
tchannel/zipkin/tracers.py:141: in __init__
    TCollectorClient = client_for('tcollector', TCollector)
tchannel/thrift/client.py:96: in client_for
    service_module, thrift_service_name, method_name
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

service_module = <module 'tchannel.zipkin.thrift.TCollector' from '/Users/blampe/src/tchannel/python/tchannel/zipkin/thrift/TCollector.pyc'>
service_name = 'TCollector', method_name = '__delattr__'

    def generate_method(service_module, service_name, method_name):
        """Generate a method for the given Thrift service.

        :param service_module:
            Thrift-generated service module
        :param service_name:
            Name of the Thrift service
        :param method_name:
            Method being called
        """
        assert service_module
        assert service_name
        assert method_name

>       args_type = getattr(service_module, method_name + '_args')
E       AttributeError: 'module' object has no attribute '__delattr___args'

tchannel/thrift/client.py:116: AttributeError
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -15,8 +15,7 @@ sudo: false
 language: python
 env:
     - TOX_ENV=py27
-    #- TOX_ENV=py34
-    #- TOX_ENV=pypy
+    - TOX_ENV=pypy
     - TOX_ENV=cover
     - TOX_ENV=flake8
 

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -4,7 +4,7 @@ Changelog
 0.10.2 (unreleased)
 -------------------
 
-- Nothing changed yet.
+- Support PyPy 2.
 
 
 0.10.1 (2015-07-10)

--- a/tchannel/sync/client.py
+++ b/tchannel/sync/client.py
@@ -133,7 +133,6 @@ class SyncClientOperation(object):
             raise gen.Return(result)
 
         future = self.threadloop.submit(make_request)
-
         return future
 
 

--- a/tchannel/sync/thrift.py
+++ b/tchannel/sync/thrift.py
@@ -20,8 +20,7 @@
 
 from __future__ import absolute_import
 
-import inspect
-
+from tchannel.thrift.util import get_service_methods
 from tchannel.thrift.client import client_for as async_client_for
 
 
@@ -69,11 +68,7 @@ def client_for(service, service_module, thrift_service_name=None):
     if not thrift_service_name:
         thrift_service_name = service_module.__name__.rsplit('.', 1)[-1]
 
-    method_names = [
-        name for (name, _) in inspect.getmembers(
-            service_module.Iface, predicate=inspect.ismethod
-        )
-    ]
+    method_names = get_service_methods(service_module.Iface)
 
     def init(self, tchannel_sync, hostport=None, trace=False):
         self.async_thrift = self.__async_client_class__(

--- a/tchannel/thrift/client.py
+++ b/tchannel/thrift/client.py
@@ -29,6 +29,7 @@ from tornado import gen
 from tchannel.tornado.broker import ArgSchemeBroker
 
 from .scheme import ThriftArgScheme
+from .util import get_service_methods
 
 # Generated clients will use this base class.
 _ClientBase = namedtuple('_ClientBase', 'tchannel hostport service trace')
@@ -72,11 +73,7 @@ def client_for(service, service_module, thrift_service_name=None):
     if not thrift_service_name:
         thrift_service_name = service_module.__name__.rsplit('.', 1)[-1]
 
-    method_names = [
-        name for (name, _) in inspect.getmembers(
-            service_module.Iface, predicate=inspect.ismethod
-        )
-    ]
+    method_names = get_service_methods(service_module.Iface)
 
     def new(cls, tchannel, hostport=None, trace=False):
         """

--- a/tchannel/thrift/util.py
+++ b/tchannel/thrift/util.py
@@ -1,0 +1,20 @@
+from __future__ import absolute_import
+
+
+import inspect
+
+
+def get_service_methods(iface):
+    """Get a list of methods defined in the interface for a Thrift service.
+
+    :param iface:
+        The Thrift-generated Iface class defining the interface for the
+        service.
+    :returns:
+        A set containing names of the methods defined for the service.
+    """
+    methods = inspect.getmembers(iface, predicate=inspect.ismethod)
+
+    return set(
+        name for (name, method) in methods if not name.startswith('__')
+    )

--- a/tests/sync/test_thrift.py
+++ b/tests/sync/test_thrift.py
@@ -41,7 +41,6 @@ def thrift_sync_client(tchannel_server, thrift_service):
 
 @pytest.mark.integration
 def test_call(tchannel_server, thrift_sync_client, thrift_service):
-
     expected = thrift_service.Item(
         key='foo', value=thrift_service.Value(integerValue=42)
     )

--- a/tests/thrift/test_util.py
+++ b/tests/thrift/test_util.py
@@ -1,0 +1,20 @@
+from __future__ import absolute_import
+
+
+from tchannel.thrift.util import get_service_methods
+
+
+def test_get_service_methods():
+
+    class Iface(object):
+
+        def __init__(self):
+            pass
+
+        def hello(self):
+            pass
+
+        def world(self, foo):
+            pass
+
+    assert set(['hello', 'world']) == get_service_methods(Iface)

--- a/tox.ini
+++ b/tox.ini
@@ -1,5 +1,5 @@
 [tox]
-envlist = py27,py34,cover,flake8,pypy
+envlist = py27,cover,flake8,pypy
 
 [testenv]
 deps =

